### PR TITLE
Fix compiler warnings with NHOP code

### PR DESCRIPTION
**Pull request by n-sandeep**
Signed-off-by: Sandeep N <sandeep.nagapattinam@intel.com>

### CODE REVIEW

**Review by nupuruttarwar — APPROVED** [review]
LGTM

File: switchapi/switch_nhop_int.h [hunk 1]
```diff
@@ -76,8 +76,8 @@ extern "C" {
 
 #define switch_nhop_get_group(_device, _handle, _info)                    \
   ({                                                                \
-    switch_nhop_info_t *_tmp_nhop_info = NULL;                      \
-    (void)(_tmp_nhop_info == *_info);                               \
+    switch_nhop_group_info_t *_tmp_nhop_group_info = NULL;          \
+    (void)(_tmp_nhop_group_info == *_info);                         \
```
Comment (ffoulkes) [inline switchapi/switch_nhop_int.h:80]: I don't understand. `(_tmp_nhop_group_info == *_info)` is a Boolean expression. Why are we casting a `bool` to `void`?

Comment (ffoulkes) [reply switchapi/switch_nhop_int.h:80]: May I suggest the following?
```
#define switch_nhop_get_group(_device, _handle, _info)                    \
    switch_handle_get(                                              \
        _device, SWITCH_HANDLE_TYPE_NHOP_GROUP, _handle, (void **)(_info))
```

**Review by ffoulkes — CHANGES_REQUESTED** [review]

**Comment by ffoulkes** [comment]
Superseded by PR #20.

**Pull request closed without merging**